homeassistant/components/vodafone_station/sensor.py

```diff
@@ -0,0 +1,217 @@
+"""Vodafone Station sensors."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import Any, Final
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, EntityCategory, UnitOfDataRate
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util.dt import utcnow
+
+from .const import _LOGGER, DOMAIN, LINE_TYPES
+from .coordinator import VodafoneStationRouter
+
+NOT_AVAILABLE: list = ["", "N/A", "0.0.0.0"]
+
+
+@dataclass
+class VodafoneStationBaseEntityDescription:
+    """Vodafone Station entity base description."""
+
+    value: Callable[[Any, Any], Any] = lambda val, key: val[key]
+    is_suitable: Callable[[dict], bool] = lambda val: True
+
+
+@dataclass
+class VodafoneStationEntityDescription(
+    VodafoneStationBaseEntityDescription, SensorEntityDescription
+):
+    """Vodafone Station entity description."""
+
+
+def _calculate_uptime(value: dict, key: str) -> datetime:
+    """Calculate device uptime."""
+    d = int(value[key].split(":")[0])
+    h = int(value[key].split(":")[1])
+    m = int(value[key].split(":")[2])
+
+    return utcnow() - timedelta(days=d, hours=h, minutes=m)
+
+
+def _line_connection(value: dict, key: str) -> str:
+    """Identify line type."""
+
+    internet_ip = value[key]
+    dsl_ip = value.get("dsl_ipaddr")
+    fiber_ip = value.get("fiber_ipaddr")
+    internet_key_ip = value.get("vf_internet_key_ip_addr")
+
+    if internet_ip == dsl_ip:
+        return LINE_TYPES[1]
+
+    if internet_ip == fiber_ip:
+        return LINE_TYPES[2]
+
+    if internet_ip == internet_key_ip:
+        return LINE_TYPES[3]
+
+    return LINE_TYPES[0]
```

Comment:
Oh, last thing. I think Unknown is already handled by HA if you return None, can you maybe check this?

Reply:
Checked and applied ;-)